backend/src/controllers/auth.controller.ts

```diff
@@ -115,3 +116,49 @@ export const logout = async (req: Request, res: Response) => {
     }
   );
 };
+
+const client = new OAuth2Client();  
+export const googleLogin = async (req: Request, res: Response) => {
+  try{
+    const ticket = await client.verifyIdToken({
+      idToken: req.body.credential,
+      audience: process.env.GOOGLE_OAUTH_CLIENT_ID,
+    });
+    const payload = ticket.getPayload();
+    const email = payload?.email;
+
+    if (!email) {
+      logger.error('Email not found in payload');
+      return res
+        .status(HttpStatusCode.BadRequest)
+        .json({ message: 'Email not found in payload' });
+    }
+
+    let user = await getUserByIdentifier(email);
+    if (!user) {
+      const newUser: User = {
+        userName: payload?.name || '',
+        email: email,
+        password: '',
```

Comment:
why is the password empty? what will happen if this user will try to login with only the username and an empty password

Reply:
moved into `auth.service.ts`.